src/Segments/VCS.hs

```diff
@@ -76,10 +80,18 @@ gitBranch = do
         Just "HEAD" -> readProcess "git" ["rev-parse", "--short", "HEAD"]
         x           -> return x
 
+gitCommitsAheadCount :: IO (Maybe String)
+gitCommitsAheadCount = runMaybeT $ do
+  branch <- MaybeT gitBranch
+  MaybeT $ (=<<) (showNZ . read) <$> readProcess "git" ["rev-list", "--count", path branch] where
+    path branch = "origin/" ++ branch ++ ".." ++ branch
```

Comment:
This line assumes that the remote is called `origin`. I think the correct way to get it would be `git config --get branch.$BRANCH.remote`

Reply:
Also, making `path` a function whose argument shadows the existing binding seems unnecessarily complex.

I think factoring out the invocation of `readProcess` into its own binding would also simplify this a bit - using `(=<<)` in prefix form feels like a bit of a smell to me.

Reply:
I refactored `gitstashCount` to follow the same pattern as `gitCommitsAheadCount`. Let me know if you rather not do it and I'll revert it.